packages/shared/src/components/modals/ReportPostModal.tsx

```diff
@@ -1,25 +1,77 @@
-import React, { ReactElement, MouseEvent, useState, useRef } from 'react';
-import { Radio } from '../fields/Radio';
+import React, {
+  ReactElement,
+  MouseEvent,
+  useState,
+  useRef,
+  useMemo,
+  SetStateAction,
+  Dispatch,
+} from 'react';
+import { Radio, RadioOption } from '../fields/Radio';
 import { Post } from '../../graphql/posts';
 import { Checkbox } from '../fields/Checkbox';
-import { Button } from '../buttons/Button';
+import { Button, ButtonSize } from '../buttons/Button';
 import { PostBootData } from '../../lib/boot';
 import { Modal, ModalProps } from './common/Modal';
+import { FlexRow } from '../utilities';
 
 export interface Props extends ModalProps {
   postIndex: number;
   post: Post | PostBootData;
-  onReport: (postIndex, postId, reason, comment, blockSource) => unknown;
+  onReport: (postIndex, postId, reason, comment, blockSource, tags) => unknown;
 }
 
 const reportReasons: { value: string; label: string }[] = [
+  { value: 'IRRELEVANT', label: 'The post is not about...' },
   { value: 'BROKEN', label: 'Broken link' },
   { value: 'CLICKBAIT', label: 'Clickbait' },
   { value: 'LOW', label: 'Low-quality content' },
   { value: 'NSFW', label: 'NSFW' },
   { value: 'OTHER', label: 'Other' },
 ];
 
+const reportReasonsMap: Partial<
+  Record<
+    string,
+    ({
+      post,
+      selectedTags,
+      setSelectedTags,
+    }: {
+      post: Post;
+      selectedTags: string[];
+      setSelectedTags: Dispatch<SetStateAction<string[]>>;
+    }) => ReactElement
+  >
+> = {
+  IRRELEVANT: ({ post, selectedTags, setSelectedTags }) => {
+    return (
+      <FlexRow className="flex-wrap gap-2 my-4">
+        {post.tags?.map((tag) => {
+          const isSelected = selectedTags.includes(tag);
+
+          return (
+            <Button
+              key={tag}
+              className={isSelected ? 'btn-primary' : 'btn-tertiaryFloat'}
+              buttonSize={ButtonSize.Small}
+              onClick={() => {
+                if (isSelected) {
+                  setSelectedTags(selectedTags.filter((t) => t !== tag));
+                } else {
+                  setSelectedTags([...selectedTags, tag]);
+                }
+              }}
+            >
+              #{tag}
+            </Button>
```

Comment:
From what I see Lee used the same in his PR, and `TagButton` seems specific to follow concept used in onboarding. Let me know if we want to adjust this.

Reply:
All good for now, maybe a future improvement we can do?
To have a global tag component that can or cannot have actions? 👀

For now this works.

Reply:
🤝